src/ai/backend/manager/scheduler/dispatcher.py

```diff
@@ -559,22 +579,40 @@ async def _schedule_multi_node_session(
             for kernel in sess_ctx.kernels:
                 try:
                     agent_alloc_ctx: AgentAllocationContext
-                    agent_id = scheduler.assign_agent_for_kernel(candidate_agents, kernel)
-                    if agent_id is None:
-                        raise InstanceNotAvailable
-
-                    async def _reserve() -> None:
-                        nonlocal agent_alloc_ctx, candidate_agents
-                        assert agent_id is not None
-                        async with agent_db_conn.begin_nested():
-                            agent_alloc_ctx = await _reserve_agent(
-                                sched_ctx, agent_db_conn,
-                                sgroup_name, agent_id, kernel.requested_slots,
-                                extra_conds=agent_query_extra_conds,
-                            )
-                            candidate_agents = await _list_agents_by_sgroup(agent_db_conn, sgroup_name)
-
-                    await execute_with_retry(_reserve)
+                    agent_id = AgentId('')
+                    if kernel.agent_id is not None:
+                        agent_id = kernel.agent_id
+                    else:
+                        agent_id = scheduler.assign_agent_for_kernel(candidate_agents, kernel)
```

Comment:
```suggestion
                    agent_id: AgentId | None
                    if kernel.agent_id is not None:
                        agent_id = kernel.agent_id
                    else:
                        agent_id = scheduler.assign_agent_for_kernel(candidate_agents, kernel)
                    assert agent_id is not None
```
Instead of assigning "arbitrary" empty value, it's better to utilize static type checker's capability to narrow down the type scope.